Functor.md

```diff
@@ -0,0 +1,340 @@
+autoscale: true
+footer: ![Arrow](arrow-brand-128x128.png) [@jorgecastillopr](https://twitter.com/jorgecastillopr) [@47deg](https://twitter.com/47deg) :: [Λrrow](http://arrow-kt.io) :: [http://arrow-kt.io/docs/typeclasses/functor/](http://arrow-kt.io/docs/typeclasses/functor/)
+slidenumbers: true
+
+# Functor
+
+__`Functor`__ is available in the __arrow-typeclasses__ module.
+
+```groovy
+// gradle
+compile 'io.arrow-kt:arrow-typeclasses:$arrowVersion'
+
+import arrow.typeclasses.Functor
+```
+
+^ The Functor is provided on the arrow-typeclasses artifact.
+^ Just add it to your build.gradle file and you'll be good to go.
+
+---
+
+# Functor
+
+`Functor` is a __Typeclass__, so it defines a given behavior.
+
+`Applicative` and `Monad` inherit its combinators. You will learn more about those in other videos in the series.
+
+Functor
+Applicative
+Monad
+
+^ Functor is defined as a Typeclass. On pure Functional Programming, Typeclasses define behaviors.
+^ We'll know about which behavior is encoded in the Functor in further slides.
+^ Other very well known typeclasses are Applicative and Monad, and this is how they're related to each other.
+(necesitaremos diagrama de cajitas con flechas de dependencia entre ellas)
+^ We'll talk about those in other Arrow videos.
+
+---
+
+# Functor :: Polymorphism
+
+```kotlin
+interface Functor<F> {
+    //...
+}
+```
+
+- Functor is __parametric over a type constructor `F`__. This allows us to build abstract functions over the behaviors that Functor defines and forget about the concrete types that F may refer to.
+
+- We call this __ad-hoc polymorphism__, the ability to write polymorphic programs that can be defined in generic terms.
+
+^ Functor is parametric over a generic type F. F here, stands for a data type or what we also call a "Type Constructor".
+^ Thanks to this, we can declare generic functions encoded on top of the behaviors provided by Functor and forget all the way about the concrete data type used for F.
+^ This is called "Ad-Hoc polymorphism" and that's how typeclasses allow us to encode completely generic and polymorphic programs that can work over many different data types.
+
+---
+
+# Functor :: map
+
+`Functor` abstracts the ability to __map__ over the computational context of a type constructor `F`.
+
+In other words, it provides a mapping function for the `F` type:
+
+```kotlin
+fun F<A>.map(f: (A) -> B): F<B>
```

Comment:
I think these should use `Kind<F, A>` instead of `F<A>` since `F<A>` is not valid kotlin syntax and will only confuse viewers.